src/transformers/trainer.py

```diff
@@ -2078,15 +2084,18 @@ def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
         if model is None:
             model = self.model
 
-        if not os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)) and not os.path.isfile(
-            os.path.join(resume_from_checkpoint, WEIGHTS_INDEX_NAME)
-        ):
+        config_file = os.path.join(resume_from_checkpoint, CONFIG_NAME)
+        weights_file = os.path.join(resume_from_checkpoint, WEIGHTS_NAME)
+        weights_index_file = os.path.join(resume_from_checkpoint, WEIGHTS_INDEX_NAME)
+        safe_weights_file = os.path.join(resume_from_checkpoint, SAFE_WEIGHTS_NAME)
```

Comment:
There could also be a sharded checkpoint (with `SAFE_WEIGHTS_INDEX_NAME`) if the model is big.

Reply:
Done. I have also added safe loading for sharded checkpoints (saving has already been implemented if I am not mistaking). Can you have a look? `load_sharded_checkpoint` at `modeling_utils.py`